xdsl/dialects/func.py

```diff
@@ -177,17 +177,11 @@ def print(self, printer: Printer):
                 printer.print(" ")
         else:
             printer.print_attribute(self.function_type)
-        attr_dict = {
-            k: v
-            for k, v in self.attributes.items()
-            if k not in ("sym_name", "function_type", "sym_visibility")
-        }
-        if len(attr_dict) > 0:
-            printer.print(" attributes {")
-            printer.print_list(
-                attr_dict.items(), lambda i: printer.print(f'"{i[0]}" = {i[1]}')
-            )
-            printer.print("}")
+
+        reserved = ("sym_name", "function_type", "sym_visibility")
+        if any(k not in reserved for k in self.attributes.keys()):
```

Comment:
Actually, can we add that to `print_op_attributes`?
We probably can do something like adding a `with_keyword` option?
Or adding a new function `print_op_attributes_with_keyword`?
We can do it in another PR if you think this is feature creep ;)

Reply:
Yup, I was wondering that! So how do you see this? I can add a `prefix` parameter defaulting to `attributes `?

Reply:
I think it makes sense to add this to the same PR tbh

Reply:
Also seems reasonable to add it to the parser also, and add a roundtrip test that the funcitonality between the two functions is somewhat mirrored

Reply:
The parser one is already there ^^'

Reply:
Yup, I wanted to add it as part of this PR and just waited for validation before wasting keystrokes 😋
Will do!

Reply:
SO yeah, I would just put a boolean rather than a string. I don't think anyone would use a different keyword than `attributes `, and we probably don't want to give the option to people (otherwise our code will become less easy to read).